Auth routes/middleware/user model

derekjou opened the pull request:
Added user model and secret key for jwt signing

derekjou committed:
User model and secret added

Added user model and secret key for jwt signing

derekjou committed:
register/login routes + auth middleware

Tentative changes to auth routes/middleware

derekjou committed:
Added path package

Omited path by accident on last push; added

derekjou committed:
removed users.js route

Removed users.js and references in app.js

derekjou committed:
switching away from babel dev

Exported env variables to be able to source locally. Start and dev commands now use node and nodemon respectively

derekjou committed:
Removed old imports

derekjou committed:
Removed passport package

ncen5293 reviewed (approved):
everything looks good

aeciorc commented on server/routes/auth.js:
```diff
@@ -0,0 +1,83 @@
+  // Use if repeat password verificatin is required
+  // if(request.password !== request.repeatPassword){
+  //     return "Passwords don't match"
+  // }
```
This one would be done on the frontend only. Backend would receive a single password

aeciorc commented on server/routes/auth.js:
```diff
@@ -0,0 +1,83 @@
+        }
+        console.log("New user created!");
+        res.send({ success: true });
+        // res.redirect("/login");
```
Let's do redirects on the frontend instead

aeciorc commented on server/routes/auth.js:
```diff
@@ -0,0 +1,83 @@
+  } catch {
+    res.status(401).send({ error: "Invalid Request" });
+  }
+}
```
I think it would be better if functions like that and the one on top are stored somewhere else, maybe in a utilities folder?

aeciorc commented on server/routes/auth.js:
```diff
@@ -0,0 +1,83 @@
+
+function jwtVerify(req, res, next) {
+  try {
+    const token = req.headers.authorization.split(" ")[1];
```
The token will be in the cookies intead

aeciorc reviewed (changes requested):
Looks good, some comments below. One other thing I suggest is having a custom error handler to avoid too much error checking logic on the routes. https://expressjs.com/en/guide/error-handling.html

derekjou committed:
Clean up errors, utils folder

Created simple error handler middleware. Created a utility folder and added validateRegistration and jwtVerify to authUtils.js

ncen5293 reviewed (approved):
everything looks good
